windows/hcsshim.go

```diff
@@ -129,7 +129,35 @@ func newWindowsContainerConfig(ctx context.Context, owner, id string, spec *spec
 		}
 		conf.MappedDirectories = mds
 	}
-
+	if r := spec.Windows.Resources; r != nil {
+		if cpu := r.CPU; cpu != nil {
+			if cpu.Count != nil {
+				conf.ProcessorCount = uint32(*cpu.Count)
+			}
+			if cpu.Shares != nil {
+				conf.ProcessorWeight = uint64(*cpu.Shares)
+			}
+			if cpu.Maximum != nil {
+				conf.ProcessorMaximum = int64(*cpu.Maximum)
+			}
+		}
+		if mem := r.Memory; mem != nil {
+			if mem.Limit != nil {
+				conf.MemoryMaximumInMB = int64(*mem.Limit)
```

Comment:
What unit is `*mem.Limit` in? I am comparing with the implementation in moby which divides by 1M.

Reply:
given the name of the variable, I'd assume this should be the value in MB (i.e. `32 * 1024 * 1024 / 1024 * 1024`). Looking at the other change in this files, `Windows.Resources.Memory.Limit` seems to be in bytes, so it looks like it does need to be divided by 1MB here.